pkg/semtech-loramac/include/semtech_loramac.h

```diff
@@ -0,0 +1,303 @@
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup         pkg_semtech-loramac
+ * @brief           Public API and definitions of the Semtech LoRaMAC
+ * @{
+ *
+ * @file
+ *
+ * @author          Alexandre Abadie <alexandre.abadie@inria.fr>
+ */
+
+#ifndef SEMTECH_LORAMAC_H
+#define SEMTECH_LORAMAC_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <inttypes.h>
+#include "net/loramac.h"
+
+#include "sx127x.h"
+
+/**
+ * @name    Definitions for messages exchanged between the MAC and call threads
+ * @{
+ */
+#define MSG_TYPE_ISR                   (0x3456)  /**< radio device ISR */
+#define MSG_TYPE_RX_TIMEOUT            (0x3457)  /**< RX timeout in the device driver */
+#define MSG_TYPE_TX_TIMEOUT            (0x3458)  /**< TX timeout in the device driver*/
+#define MSG_TYPE_MAC_TIMEOUT           (0x3459)  /**< MAC timers timeout */
+#define MSG_TYPE_LORAMAC_CMD           (0x3460)  /**< Command sent to the MAC */
+#define MSG_TYPE_LORAMAC_NOTIFY        (0x3461)  /**< MAC notifications */
+/** @} */
+
+/**
+ * @brief   Maximum payload size of a LoRaWAN application data
+ */
+#define LORAWAN_APP_DATA_MAX_SIZE      (242U)
+
+/**
+ * @brief   LoRaMAC status
+ */
+enum {
+    SEMTECH_LORAMAC_JOIN_SUCCEEDED,              /**< Join procedure succeeded */
+    SEMTECH_LORAMAC_JOIN_FAILED,                 /**< Join procedure failed */
+    SEMTECH_LORAMAC_NOT_JOINED,                  /**< MAC is not joined */
+    SEMTECH_LORAMAC_TX_DONE,                     /**< Transmission completed */
+    SEMTECH_LORAMAC_RX_DATA,                     /**< Data received */
+};
+
+typedef struct {
+    uint8_t payload[LORAWAN_APP_DATA_MAX_SIZE];  /**< RX payload buffer */
+    uint8_t payload_len;                         /**< Length of the RX payload */
+    uint8_t port;                                /**< RX port */
+} semtech_loramac_rx_data_t;
+
+/**
+ * @brief   Initializes semtech loramac
+ *
+ * @param[in] dev          pointer to the radio device
+ *
+ * @return                 0 on success
+ * @return                -1 on failure
+ */
+int semtech_loramac_init(sx127x_t *dev);
+
+/**
+ * @brief   Starts a LoRaWAN network join procedure
+ *
+ * @param[in] type         The type of join procedure (otaa or abp)
+ *
+ * @return                 SEMTECH_LORAMAC_JOIN_SUCCEEDED on success
+ * @return                 SEMTECH_LORAMAC_JOIN_FAILED on failure
+ */
+uint8_t semtech_loramac_join(uint8_t type);
+
+/**
+ * @brief   Sends data to LoRaWAN
+ *
+ * @param[in] cnf          Use confirmable/unconfirmable send type
+ * @param[in] port         The send port to use (between 1 and 223 for application)
+ * @param[in] tx_buf       The TX buffer
+ * @param[in] tx_len       The length of the TX buffer
+ * @param[out] rx_data     The RX data descriptor
+ *
+ * @return                 SEMTECH_LORAMAC_NOT_JOINED when the network is not joined
```

Comment:
Here and everywhere else: lines are too long. I propose maybe changing the alignment after all.